go/vt/vtgate/planbuilder/operators/join_merging.go

```diff
@@ -76,6 +83,28 @@ func mergeJoinInputs(ctx *plancontext.PlanningContext, lhs, rhs Operator, joinPr
 	}
 }
 
+func mergeAnyShardRoutings(ctx *plancontext.PlanningContext, a Routing, b Routing, joinPredicates []sqlparser.Expr, joinType sqlparser.JoinType) *AnyShardRouting {
+	ar := a.(*AnyShardRouting)
+	br := b.(*AnyShardRouting)
```

Comment:
nit: might be better to set the type of `a` and `b` to `*AnyShardRouting` directly, do the cast in the caller, not leaving that responsibility to the function itself.